src/Monarch/Queue.purs

```diff
@@ -38,3 +40,16 @@ new = do
       pure $ delete next
     dispatch x = Ref.read nexts >>= traverse_ (_ $ x)
   pure { event, dispatch }
+
+shareReplayLast :: forall a . Event a -> Effect (Event a)
+shareReplayLast source = do
+  lastValueRef <- Ref.new Nothing
+  sink <- new
+  unsubscribeSource <- source # subscribe \x -> do
+    sink.dispatch x
+    Ref.write (Just x) lastValueRef
+  pure $ Event \next -> do
+    Ref.read lastValueRef >>= whenJust next
+    unsubscribeSink <- sink.event # subscribe next
+    -- FIXME: unsubscribe from the source event
```

Comment:
Is there any way source event can be unsubscribed? One purpose of this function is to always have latest value but it can't have latest value when unsubscribed from source event.

Reply:
I was thinking about it. :/
An idea could be reference counting for the listeners and in the unsubscription of the sinking event, we can unsubscribe from the source event when there is no other listener to the sinking event. For the subscribers of the sinking event, we will subscribe to the source event another time.